From bcebf6f6056651f27423e47b9b04b1ee3e282984 Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Wed, 12 Jul 2023 21:10:35 +0300
Subject: [PATCH 19/81] Build with QT_NO_CONTEXTLESS_CONNECT

This amends commit c641e462e2bd33972646bd20bc76f8dff4e6d01d.

This disables the 3-arg QObject::connect() overload:
QObject::connect(sender, signal, functor)

For details see:
https://lists.qt-project.org/pipermail/development/2023-July/044141.html

No changes required.

Change-Id: I2a351e30ffcaae85e874ec344f4daa00ed471b46
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
---
 src/plugins/iconengines/svgiconengine/CMakeLists.txt | 2 ++
 src/plugins/imageformats/svg/CMakeLists.txt          | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/plugins/iconengines/svgiconengine/CMakeLists.txt b/src/plugins/iconengines/svgiconengine/CMakeLists.txt
index 7e4c4a6..3caa9de 100644
--- a/src/plugins/iconengines/svgiconengine/CMakeLists.txt
+++ b/src/plugins/iconengines/svgiconengine/CMakeLists.txt
@@ -11,6 +11,8 @@ qt_internal_add_plugin(QSvgIconPlugin
     SOURCES
         main.cpp
         qsvgiconengine.cpp qsvgiconengine.h
+    DEFINES
+        QT_NO_CONTEXTLESS_CONNECT
     LIBRARIES
         Qt::Core
         Qt::CorePrivate
diff --git a/src/plugins/imageformats/svg/CMakeLists.txt b/src/plugins/imageformats/svg/CMakeLists.txt
index 140e281..8091498 100644
--- a/src/plugins/imageformats/svg/CMakeLists.txt
+++ b/src/plugins/imageformats/svg/CMakeLists.txt
@@ -11,6 +11,8 @@ qt_internal_add_plugin(QSvgPlugin
     SOURCES
         main.cpp
         qsvgiohandler.cpp qsvgiohandler.h
+    DEFINES
+        QT_NO_CONTEXTLESS_CONNECT
     LIBRARIES
         Qt::Core
         Qt::Gui
-- 
2.42.1

